blockchain/blockchain.go

```diff
@@ -734,6 +734,17 @@ func revertHead(txn db.Transaction) error {
 	if err != nil {
 		return err
 	}
+
+	// Don't revert part of the L1-verified chain.
+	// If the L1 head isn't in the db yet, revert.
+	// If the L2 height is less than the L1 height, don't revert.
+	var head *core.L1Head
+	if head, err = l1Head(txn); err != nil && !errors.Is(err, db.ErrKeyNotFound) {
+		return err
+	} else if head != nil && blockNumber <= head.BlockNumber {
```

Comment:
This assumes that our L1 head always agrees with our L2 chain. I don't think we ensure that yet, right?

Reply:
Correct. I think when I first wrote the PR I intended to do the changes separately, but can't remember.

In any case, the latest changes should ensure the L1 and L2 chains always agree.